alg/gdalgrid.cpp

```diff
@@ -2646,10 +2912,134 @@ CPLErr ParseAlgorithmAndOptions( const char *pszAlgorithm,
             break;
         }
         case GGA_MetricMinimum:
+        {
```

Comment:
that block as well as for the other metric constants seem to be a copy&paste of the generic one that already existed and applied to all of them. Why is that needed ?

Reply:
@rouault Good point. Did not see the default in the switch statement. Is indeed not needed!

Reply:
what about changing the switch statement so that the default case contains everything of the current GGA_MetricMinimum  case? Otherwise, the GGA_Metric* cases would include stuff like power, smoothing and max_points, which are not needed in those cases.

Reply:
> what about changing the switch statement 

I would just revert those changes. All the GGA_Metric* cases will go to the "case GGA_MetricAverageDistancePts:" which does the right thing